controllers/metal3.io/host_state_machine.go

```diff
@@ -293,16 +293,25 @@ func (hsm *hostStateMachine) provisioningCancelled() bool {
 	if hsm.Host.HasError() {
 		return true
 	}
-	if hsm.Host.Spec.Image == nil {
+	if hsm.Host.Spec.Image == nil && hsm.Host.Spec.LiveImage == nil {
```

Comment:
This logic is very similar to the BMH NeedsProvisioning() - I wonder if we could use the inverse of that here instead of duplicating nearly the same conditionals?